modules/tinymce/src/core/test/ts/browser/content/EditorContentTest.ts

```diff
@@ -80,28 +91,20 @@ describe('browser.tinymce.core.content.EditorContentTest', () => {
     assertEventsContentType();
   });
 
-  it('TINY-6281: getContent text', () => {
-    const editor = hook.editor();
-    editor.setContent('<p>Text to be retrieved</p>');
-    const text = editor.getContent({ format: 'text' });
-    // TODO: TINY-8367 The table plugin code has been moved to core so there is now always an extra DIV in the dom for the table resize bars
-    // Safari differs in behaviour compared to the other browsers when getting text content for inline mode.
-    // When the resize bar div is included in the DOM, editor.getBody().innerText includes two extra \n at the end
-    const isSafari = PlatformDetection.detect().browser.isSafari();
-    const expected = 'Text to be retrieved' + (isSafari ? '\n\n' : '');
-    Assertions.assertHtml('Should be expected text', expected, text);
-    assertEventsFiredInOrder();
-    assertEventsContentType();
-  });
+  // TODO: TINY-8367 The table plugin code has been moved to core so there is now always an extra DIV in the dom for the table resize bars
+  // Safari differs in behaviour compared to the other browsers when getting text content for inline mode.
+  // When the resize bar div is included in the DOM, editor.getBody().innerText includes two extra \n at the end
+  it('TINY-6281: getContent text', () => testGetTextContent('<p>Text to be retrieved</p>', 'Text to be retrieved' + (isSafari() ? '\n\n' : '')));
 
-  it('TINY-6281: getContent text with empty editor', () => {
-    const editor = hook.editor();
-    editor.setContent('');
-    const text = editor.getContent({ format: 'text' });
-    Assertions.assertHtml('Should be expected text', '', text);
-    assertEventsFiredInOrder();
-    assertEventsContentType();
-  });
+  it('TINY-8578: getContent text, empty line in div', () => testGetTextContent('<div><p></p></div>', ''));
+
+  it('TINY-8578: getContent text, empty line', () => testGetTextContent('<p></p>', ''));
+
+  it('TINY-8578: getContent text, two empty lines in div', () => testGetTextContent('<div><p></p><p></p></div>', isSafari() ? '\n\n' : '\n\n\n\n'));
```

Comment:
With regards to this difference, it appears that Safari only counts `<p><br></p>` as a single newline, instead of 2 newlines unlike other browsers.